featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -194,18 +194,8 @@ def __init__(
 
         self.ignore_columns = defaultdict(set)
         if ignore_columns is not None:
-            # check if ignore_columns is not {str: list}
-            if not all(isinstance(i, str) for i in ignore_columns.keys()) or not all(
-                isinstance(i, list) for i in ignore_columns.values()
-            ):
-                raise TypeError("ignore_columns should be dict[str -> list]")
-            # check if list values are all of type str
-            elif not all(
-                all(isinstance(v, str) for v in value)
-                for value in ignore_columns.values()
-            ):
-                raise TypeError("list values should be of type str")
             for df_name, cols in ignore_columns.items():
```

Comment:
What about refactoring this line and the next two into its own function `_make_ignore_columns` (or something similar). It would help us separate implementation details from the algorithm itself. 

More generally, I am curious to hear thoughts about refactoring chunks like 

``` 
self.agg_primitives = sorted(
            [
                check_primitive(
                    p,
                    "aggregation",
                    aggregation_primitive_dict,
                    transform_primitive_dict,
                )
                for p in agg_primitives
            ],
        )
```

into their own functions? We could then have a more testable API that might be easier to reason about (i.e. we could reason about what we want `self.agg_primitives` to be rather than how do it).

Reply:
Yeah, any consolidation of duplicated code is a win in my book, what you have here already is a good step in making our code more testable, readable, and efficient. Also, I prefer to keep mrs small so if we go down the path of refactoring something else let us do it in a separate mr.

Reply:
I agree. I will create an issue to track that refactoring.